File: src/App/entity/Patient.php
<?php

namespace App\entity;

use DateTime;

class Patient
{
    private ?int $id;
    private string $patientCode;
    private string $lastname;
    private string $firstname;
    private string $address;
    private string $phone;
    private array $medicalHistory;
    private string $email;
    private string $password;
    private DateTime $birthday;
    private DateTime $creatAt;
    private array $demandes;

    public function __construct(
        int | null $id = null,
        string | null $patientCode = null,
        string $lastname,
        string $firstname,
        string | null $address = null,
        string | null $phone = null,
        array | null $medicalHistory = null,
        string $email,
        string $password,
        DateTime $birthday,
        DateTime | null $creatAt = null,
        array | null $demandes = null
    )
    {
        $this->id = $id;
        $this->patientCode = $patientCode;
        $this->lastname = $lastname;
        $this->firstname = $firstname;
        $this->address = $address;
        $this->phone = $phone;
        $this->medicalHistory = $medicalHistory; 
        $this->email = $email;
        $this->password = $password; 
        $this->birthday = $birthday;
        $this->creatAt = $creatAt;
        $this->demandes = $demandes;
    }

    public function getId(): int
    {
        return $this->id;
    }

    public function getPatientCode(): string
    {
        return $this->patientCode;
    }
    
    public function getLastname(): string
    {
        return $this->lastname;
    }

    public function getFirstname(): string
    {
        return $this->firstname;
    }
    
    public function getAddress(): string
    {
        return $this->address;
    }

    public function getPhone(): string
    {
        return $this->phone;
    }
    
    public function getMedicalHistory(): array
    {
        return $this->medicalHistory;
    }

    public function getEmail(): string
    {
        return $this->email;
    }
    
    public function getPassword(): string
    {
        return $this->password;
    }

    public function getBirthday(): DateTime
    {
        return $this->birthday;
    }
    
    public function getCreatAt(): DateTime
    {
        return $this->creatAt;
    }
    
    public function getDemandes(): array
    {
        return $this->demandes;
    }

    public function setId(int $id): void
    {
        $this->id = $id;
    }

    public function setPatientCode(string $patientCode): void
    {
        $this->patientCode = $patientCode;
    }
    
    public function setLastname(string $lastname): void
    {
        $this->lastname = $lastname;
    }

    public function setFirstname(string $firstname): void
    {
        $this->firstname = $firstname;
    }
    
    public function setAddress(string $address): void
    {
        $this->address = $address;
    }

    public function setPhone(string $phone): void
    {
        $this->phone = $phone;
    }
    
    
    public function setMedicalHistory(array $medicalHistory): void
    {
        $this->medicalHistory = $medicalHistory;
    }

    public function setEmail(string $email): void
    {
        $this->email = $email;
    }
    
    public function setPassword(string $password): void
    {
        $this->password = $password;
    }

    public function setBirthday(DateTime $birthday): void
    {
        $this->birthday = $birthday;
    }
    
    public function setCreatAt(DateTime $creatAt): void
    {
        $this->creatAt = $creatAt;
    }

    public function setDemandes(array $demandes): void
    {
        $this->demandes = $demandes;
    }

    
    /**
     * @return string
    */

    public function __toString(): string
    {
        return $this->lastname . ' ' . $this->firstname . ' ' . $this->address . ' ' . $this->phone . ' ' . $this->medicalHistory . ' ' . $this->email . ' ' . $this->password . ' ' . $this->birthday . ' ' . $this->creatAt . ' ' . $this->demandes;
    }
}

File: rdvp/src/main/java/com/grdvp/entity/Patient.java
package com.grdvp.entity;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

public class Patient {
    private Integer id; //Generer automatiquement dans la base de donnees
    private String patientCode; //Generer automatiquement dans le service
    private String lastname; //Obligatoire lors de la creation d'un nouveau patient
    private String firstname; //Obligatoire lors de la creation d'un nouveau patient
    private String address; //Facultatif lors de la creation d'un nouveau patient
    private String phone;  //Obligatoire lors de la creation d'un nouveau patient
    private List<String> medicalHistory; //Facultatif lors de la creation d'un nouveau patient
    private String email; //Obligatoire lors de la creation d'un nouveau patient
    private String password; //Obligatoire lors de la creation d'un nouveau patient
    private LocalDate birthday; //Obligatoire lors de la creation d'un nouveau patient
    private LocalDateTime createdAt; //Generer automatiquement dans le service || dans la base de donnees
    private List<DemandeRDV> demandes;

    public Patient(
            Integer id,
            String patientCode,
            String lastname,
            String firstname,
            String address,
            String phone,
            List<String> medicalHistory,
            String email,
            String password,
            LocalDate birthday,
            LocalDateTime createdAt,
            List<DemandeRDV> demandes
    ) {
        this.id = id;
        this.patientCode = patientCode;
        this.lastname = lastname;
        this.firstname = firstname;
        this.address = address;
        this.phone = phone;
        this.medicalHistory = medicalHistory != null ? medicalHistory : new ArrayList<>();
        this.email = email;
        this.password = password;
        this.birthday = birthday;
        this.createdAt = createdAt;
        this.demandes = demandes;
    }

    public Patient(
            String lastname,
            String firstname,
            //String address,
            String phone,
            String email,
            String password
            //LocalDate birthday
    ) {
        //this.id = id;
        //this.patientCode = patientCode;
        this.lastname = lastname;
        this.firstname = firstname;
        //this.address = address;
        this.phone = phone;
        //this.medicalHistory = medicalHistory != null ? medicalHistory : new ArrayList<>();
        this.email = email;
        this.password = password;
        //this.birthday = birthday;
        //this.createdAt = createdAt;
        //this.demandes = demandes;
    }

    public Patient() {
        this.medicalHistory = new ArrayList<>();
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getPatientCode() {
        return patientCode;
    }

    public void setPatientCode(String patientCode) {
        this.patientCode = patientCode;
    }

    public String getLastname() {
        return lastname;
    }

    public void setLastname(String lastname) {
        this.lastname = lastname;
    }

    public String getFirstname() {
        return firstname;
    }

    public void setFirstname(String firstname) {
        this.firstname = firstname;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public List<String> getMedicalHistory() {
        return medicalHistory;
    }

    public void setMedicalHistory(List<String> medicalHistory) {
        this.medicalHistory = medicalHistory;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public LocalDate getBirthday() {
        return birthday;
    }

    public void setBirthday(LocalDate birthday) {
        this.birthday = birthday;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public List<DemandeRDV> getDemandes() {
        return demandes;
    }


    
    public String toString() {
        return String.format(
            "%s %s %s %s %s %s %s %s %s %s %s",
            Objects.toString(lastname, ""),
            Objects.toString(firstname, ""),
            Objects.toString(address, ""),
            Objects.toString(phone, ""),
            Objects.toString(medicalHistory, ""),
            Objects.toString(email, ""),
            Objects.toString(password, ""),
            Objects.toString(birthday, ""),
            Objects.toString(createdAt, ""),
            Objects.toString(demandes, ""),
            Objects.toString(patientCode, "")
        );
    }
}